ocamlPackages.janeStreet: 0.14 -> 0.15

bcc32 opened the pull request:
###### Description of changes

- Major breaking change: the library Core_kernel is no more.  The opam packages core and core_kernel have been shuffled around, so core is now platform-independent, and the Unix-dependent parts are now in core_unix.  See https://discuss.ocaml.org/t/goodbye-core-kernel/8393
- Package sexp_diff_kernel was renamed sexp_diff.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

anmonteiro commented:
Can also confirm packages build, as evidenced by e.g. [this GH Actions run](https://github.com/anmonteiro/nix-overlays/runs/6106220718?check_suite_focus=true).

The only error I ran into was the (already known and reported) https://github.com/janestreet/core_unix/issues/2.

bcc32 committed:
Add Jane Street ocaml packages version 0.15

- ocamlPackages.tls*: 0.15.2 -> 0.15.3
- ocamlPackages.bistro: unstable-2021-11-13 -> unstable-2022-05-07
- ocamlPackages.phylogenetics: 0.1.0 -> unstable-2022-05-06

symphorien commented:
there are some large committed patches. Did you consider submitting them upstream?


> The only error I ran into was the (already known and reported) https://github.com/janestreet/core_unix/issues/2.

looks like it was fixed upstream by https://github.com/janestreet/jst-config/commit/e5fdac6e5df9ba93e014a4d2db841fdbf209446f

symphorien commented:
@ofborg eval

symphorien committed:
ocamlPackages.ppx_css: fix eval

sedlex is already 2.5

symphorien committed:
ocamlPackages.jst-config: fetch upstream patch

https://github.com/NixOS/nixpkgs/pull/166033#issuecomment-1104695149

symphorien commented:
I pushed a commit pulling the fix you pointed out, @anmonteiro 

@bcc32 you marked this PR as draft. As far as I tested (nix-review) it seems fine; is there something missing or can I merge it?

bcc32 commented:
> @bcc32 you marked this PR as draft. As far as I tested (nix-review) it seems fine; is there something missing or can I merge it?

It's marked as a draft since it still includes a couple of large patches (especially `comby.patch`).

I suppose I'd like some advice here as a still-somewhat-new package maintainer.  Maybe the right thing to do with some of these _enormous_ patches is to just make the downstream package use version 0.14 for now, and make separate PRs upgrading them later on?  That way we can get this PR merged soon but without introducing a 1k-line patch file.

symphorien commented:
ligo also looks like a leaf package.

bcc32 commented:
Thanks, that makes sense.  I'll leave the patches in for now (it's not obvious to me what the cleanest way is to make ligo and comby use janestreet 0.14).

ulrikstrid commented:
@bcc32 I can try to upstream the upgrade of janestreet packages to ligo since I'm kind of working with them. The sad part is that we'll need a bunch of other upgrades in nixpkgs to upstream it  afterwards because of the upgraded tezos version.

siraben commented:
Result of `nixpkgs-review pr 166033` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>flitter</li>
    <li>ocamlPackages.ocaml-r</li>
    <li>ocamlPackages.reason-native.qcheck-rely</li>
    <li>ocamlPackages.torch</li>
    <li>ocamlPackages.wodan</li>
    <li>ocamlPackages.wodan-irmin</li>
    <li>ocamlPackages.wodan-unix</li>
  </ul>
</details>
<details>
  <summary>49 packages failed to build:</summary>
  <ul>
    <li>coqPackages.serapi</li>
    <li>haxe</li>
    <li>haxePackages.format</li>
    <li>haxePackages.heaps</li>
    <li>haxePackages.hlopenal</li>
    <li>haxePackages.hlsdl</li>
    <li>haxePackages.hxcpp</li>
    <li>haxePackages.hxcs</li>
    <li>haxePackages.hxjava</li>
    <li>haxePackages.hxnodejs_4</li>
    <li>jackline</li>
    <li>ocamlPackages.arp</li>
    <li>ocamlPackages.async_rpc_websocket</li>
    <li>ocamlPackages.bonsai</li>
    <li>ocamlPackages.caqti-driver-mariadb</li>
    <li>ocamlPackages.carton</li>
    <li>ocamlPackages.carton-git</li>
    <li>ocamlPackages.carton-lwt</li>
    <li>ocamlPackages.cohttp-async</li>
    <li>ocamlPackages.cohttp-mirage</li>
    <li>ocamlPackages.cohttp_async_websocket</li>
    <li>ocamlPackages.conduit-mirage</li>
    <li>ocamlPackages.dns-certify</li>
    <li>ocamlPackages.dns-cli</li>
    <li>ocamlPackages.dns-client</li>
    <li>ocamlPackages.dns-mirage</li>
    <li>ocamlPackages.dns-resolver</li>
    <li>ocamlPackages.dns-server</li>
    <li>ocamlPackages.dns-stub</li>
    <li>ocamlPackages.git</li>
    <li>ocamlPackages.git-cohttp</li>
    <li>ocamlPackages.git-cohttp-unix</li>
    <li>ocamlPackages.git-paf</li>
    <li>ocamlPackages.git-unix</li>
    <li>ocamlPackages.irmin-containers</li>
    <li>ocamlPackages.irmin-git</li>
    <li>ocamlPackages.irmin-http</li>
    <li>ocamlPackages.irmin-mirage-git</li>
    <li>ocamlPackages.irmin-mirage-graphql</li>
    <li>ocamlPackages.irmin-unix</li>
    <li>ocamlPackages.mirage-nat</li>
    <li>ocamlPackages.mrmime</li>
    <li>ocamlPackages.paf</li>
    <li>ocamlPackages.paf-cohttp</li>
    <li>ocamlPackages.paf-le</li>
    <li>ocamlPackages.piaf</li>
    <li>ocamlPackages.rosetta</li>
    <li>ocamlPackages.tcpip</li>
    <li>wayland-proxy-virtwl</li>
  </ul>
</details>
<details>
  <summary>400 packages built:</summary>
  <ul>
    <li>acgtk</li>
    <li>beluga</li>
    <li>comby</li>
    <li>coqPackages.coq-elpi</li>
    <li>coqPackages.graph-theory</li>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.mathcomp-analysis</li>
    <li>coqPackages.trakt</li>
    <li>dune-release</li>
    <li>easycrypt</li>
    <li>eff</li>
    <li>flow</li>
    <li>framac</li>
    <li>fstar</li>
    <li>libbap</li>
    <li>ocaml-crunch</li>
    <li>ocamlPackages.abstract_algebra</li>
    <li>ocamlPackages.accessor</li>
    <li>ocamlPackages.accessor_async</li>
    <li>ocamlPackages.accessor_base</li>
    <li>ocamlPackages.accessor_core</li>
    <li>ocamlPackages.alcotest-lwt</li>
    <li>ocamlPackages.alcotest-mirage</li>
    <li>ocamlPackages.angstrom</li>
    <li>ocamlPackages.angstrom-async</li>
    <li>ocamlPackages.angstrom-lwt-unix</li>
    <li>ocamlPackages.angstrom-unix</li>
    <li>ocamlPackages.asn1-combinators</li>
    <li>ocamlPackages.async</li>
    <li>ocamlPackages.async_extra</li>
    <li>ocamlPackages.async_find</li>
    <li>ocamlPackages.async_interactive</li>
    <li>ocamlPackages.async_js</li>
    <li>ocamlPackages.async_kernel</li>
    <li>ocamlPackages.async_rpc_kernel</li>
    <li>ocamlPackages.async_sendfile</li>
    <li>ocamlPackages.async_shell</li>
    <li>ocamlPackages.async_ssl</li>
    <li>ocamlPackages.async_unix</li>
    <li>ocamlPackages.async_websocket</li>
    <li>ocamlPackages.awa</li>
    <li>ocamlPackages.awa-lwt</li>
    <li>ocamlPackages.awa-mirage</li>
    <li>ocamlPackages.bap</li>
    <li>ocamlPackages.base</li>
    <li>ocamlPackages.base64</li>
    <li>ocamlPackages.base_bigstring</li>
    <li>ocamlPackages.base_quickcheck</li>
    <li>ocamlPackages.bigarray-overlap</li>
    <li>ocamlPackages.bignum</li>
    <li>ocamlPackages.bin_prot</li>
    <li>ocamlPackages.biocaml</li>
    <li>ocamlPackages.bisect_ppx</li>
    <li>ocamlPackages.bistro</li>
    <li>ocamlPackages.bls12-381-unix</li>
    <li>ocamlPackages.bos</li>
    <li>ocamlPackages.brisk-reconciler</li>
    <li>ocamlPackages.ca-certs</li>
    <li>ocamlPackages.ca-certs-nss</li>
    <li>ocamlPackages.camlimages</li>
    <li>ocamlPackages.caqti</li>
    <li>ocamlPackages.caqti-async</li>
    <li>ocamlPackages.caqti-driver-postgresql</li>
    <li>ocamlPackages.caqti-driver-sqlite3</li>
    <li>ocamlPackages.caqti-dynload</li>
    <li>ocamlPackages.caqti-lwt</li>
    <li>ocamlPackages.caqti-type-calendar</li>
    <li>ocamlPackages.cfstream</li>
    <li>ocamlPackages.checkseum</li>
    <li>ocamlPackages.cinaps</li>
    <li>ocamlPackages.cohttp</li>
    <li>ocamlPackages.cohttp-lwt</li>
    <li>ocamlPackages.cohttp-lwt-unix</li>
    <li>ocamlPackages.conduit</li>
    <li>ocamlPackages.conduit-async</li>
    <li>ocamlPackages.conduit-lwt</li>
    <li>ocamlPackages.conduit-lwt-unix</li>
    <li>ocamlPackages.containers</li>
    <li>ocamlPackages.containers-data</li>
    <li>ocamlPackages.cooltt</li>
    <li>ocamlPackages.core</li>
    <li>ocamlPackages.core_bench</li>
    <li>ocamlPackages.core_extended</li>
    <li>ocamlPackages.core_kernel</li>
    <li>ocamlPackages.core_unix</li>
    <li>ocamlPackages.cow</li>
    <li>ocamlPackages.crunch</li>
    <li>ocamlPackages.cstruct-async</li>
    <li>ocamlPackages.cstruct-sexp</li>
    <li>ocamlPackages.csvfields</li>
    <li>ocamlPackages.dap</li>
    <li>ocamlPackages.data-encoding</li>
    <li>ocamlPackages.dbf</li>
    <li>ocamlPackages.decompress</li>
    <li>ocamlPackages.delimited_parsing</li>
    <li>ocamlPackages.digestif</li>
    <li>ocamlPackages.dns</li>
    <li>ocamlPackages.dns-tsig</li>
    <li>ocamlPackages.dose3</li>
    <li>ocamlPackages.dune-rpc</li>
    <li>ocamlPackages.ecaml</li>
    <li>ocamlPackages.eliom</li>
    <li>ocamlPackages.elpi</li>
    <li>ocamlPackages.email_message</li>
    <li>ocamlPackages.emile</li>
    <li>ocamlPackages.encore</li>
    <li>ocamlPackages.erm_xmpp</li>
    <li>ocamlPackages.ethernet</li>
    <li>ocamlPackages.expect_test_helpers_async</li>
    <li>ocamlPackages.expect_test_helpers_core</li>
    <li>ocamlPackages.ezjsonm</li>
    <li>ocamlPackages.faraday-async</li>
    <li>ocamlPackages.fieldslib</li>
    <li>ocamlPackages.flex</li>
    <li>ocamlPackages.functoria</li>
    <li>ocamlPackages.functoria-runtime</li>
    <li>ocamlPackages.fuzzy_match</li>
    <li>ocamlPackages.gapi_ocaml</li>
    <li>ocamlPackages.gen_js_api</li>
    <li>ocamlPackages.genspio</li>
    <li>ocamlPackages.graphql-cohttp</li>
    <li>ocamlPackages.graphql_ppx</li>
    <li>ocamlPackages.h2</li>
    <li>ocamlPackages.hack_parallel</li>
    <li>ocamlPackages.happy-eyeballs</li>
    <li>ocamlPackages.higher_kinded</li>
    <li>ocamlPackages.higlo</li>
    <li>ocamlPackages.hpack</li>
    <li>ocamlPackages.httpaf</li>
    <li>ocamlPackages.imagelib</li>
    <li>ocamlPackages.incr_dom</li>
    <li>ocamlPackages.incr_map</li>
    <li>ocamlPackages.incr_select</li>
    <li>ocamlPackages.incremental</li>
    <li>ocamlPackages.index</li>
    <li>ocamlPackages.int_repr</li>
    <li>ocamlPackages.ipaddr</li>
    <li>ocamlPackages.ipaddr-cstruct</li>
    <li>ocamlPackages.ipaddr-sexp</li>
    <li>ocamlPackages.iri</li>
    <li>ocamlPackages.irmin</li>
    <li>ocamlPackages.irmin-chunk</li>
    <li>ocamlPackages.irmin-fs</li>
    <li>ocamlPackages.irmin-graphql</li>
    <li>ocamlPackages.irmin-layers</li>
    <li>ocamlPackages.irmin-mirage</li>
    <li>ocamlPackages.irmin-pack</li>
    <li>ocamlPackages.irmin-test</li>
    <li>ocamlPackages.irmin-watcher</li>
    <li>ocamlPackages.iter</li>
    <li>ocamlPackages.jane-street-headers</li>
    <li>ocamlPackages.jingoo</li>
    <li>ocamlPackages.js_of_ocaml</li>
    <li>ocamlPackages.js_of_ocaml-compiler</li>
    <li>ocamlPackages.js_of_ocaml-lwt</li>
    <li>ocamlPackages.js_of_ocaml-ppx</li>
    <li>ocamlPackages.js_of_ocaml-ppx_deriving_json</li>
    <li>ocamlPackages.js_of_ocaml-tyxml</li>
    <li>ocamlPackages.json-data-encoding</li>
    <li>ocamlPackages.json-data-encoding-bson</li>
    <li>ocamlPackages.jsonaf</li>
    <li>ocamlPackages.jst-config</li>
    <li>ocamlPackages.junit</li>
    <li>ocamlPackages.junit_alcotest</li>
    <li>ocamlPackages.junit_ounit</li>
    <li>ocamlPackages.jwto</li>
    <li>ocamlPackages.lens</li>
    <li>ocamlPackages.letsencrypt</li>
    <li>ocamlPackages.letsencrypt-app</li>
    <li>ocamlPackages.letsencrypt-dns</li>
    <li>ocamlPackages.logs</li>
    <li>ocamlPackages.lsp</li>
    <li>ocamlPackages.lwt-exit</li>
    <li>ocamlPackages.lwt_ppx</li>
    <li>ocamlPackages.macaddr</li>
    <li>ocamlPackages.macaddr-cstruct</li>
    <li>ocamlPackages.macaddr-sexp</li>
    <li>ocamlPackages.macaque</li>
    <li>ocamlPackages.mdx</li>
    <li>ocamlPackages.metrics-lwt</li>
    <li>ocamlPackages.metrics-mirage</li>
    <li>ocamlPackages.metrics-unix</li>
    <li>ocamlPackages.mimic</li>
    <li>ocamlPackages.mirage</li>
    <li>ocamlPackages.mirage-block-combinators</li>
    <li>ocamlPackages.mirage-block-unix</li>
    <li>ocamlPackages.mirage-bootvar-xen</li>
    <li>ocamlPackages.mirage-channel</li>
    <li>ocamlPackages.mirage-crypto-ec</li>
    <li>ocamlPackages.mirage-crypto-pk</li>
    <li>ocamlPackages.mirage-crypto-rng</li>
    <li>ocamlPackages.mirage-crypto-rng-async</li>
    <li>ocamlPackages.mirage-crypto-rng-mirage</li>
    <li>ocamlPackages.mirage-flow-combinators</li>
    <li>ocamlPackages.mirage-flow-unix</li>
    <li>ocamlPackages.mirage-logs</li>
    <li>ocamlPackages.mirage-net</li>
    <li>ocamlPackages.mirage-net-xen</li>
    <li>ocamlPackages.mirage-profile</li>
    <li>ocamlPackages.mirage-protocols</li>
    <li>ocamlPackages.mirage-runtime</li>
    <li>ocamlPackages.mirage-stack</li>
    <li>ocamlPackages.mirage-types</li>
    <li>ocamlPackages.mirage-types-lwt</li>
    <li>ocamlPackages.mirage-unix</li>
    <li>ocamlPackages.mirage-vnetif</li>
    <li>ocamlPackages.mirage-xen</li>
    <li>ocamlPackages.mldoc</li>
    <li>ocamlPackages.mtime</li>
    <li>ocamlPackages.mustache</li>
    <li>ocamlPackages.netchannel</li>
    <li>ocamlPackages.ocaml-compiler-libs</li>
    <li>ocamlPackages.ocaml-embed-file</li>
    <li>ocamlPackages.ocaml-lsp</li>
    <li>ocamlPackages.ocaml-monadic</li>
    <li>ocamlPackages.ocaml-print-intf</li>
    <li>ocamlPackages.ocaml_intrinsics</li>
    <li>ocamlPackages.ocamlformat-rpc-lib</li>
    <li>ocamlPackages.ocf_ppx</li>
    <li>ocamlPackages.ocplib-json-typed</li>
    <li>ocamlPackages.ocplib-json-typed-browser</li>
    <li>ocamlPackages.ocplib-json-typed-bson</li>
    <li>ocamlPackages.ocsigen-ppx-rpc</li>
    <li>ocamlPackages.ocsigen-start</li>
    <li>ocamlPackages.ocsigen-toolkit</li>
    <li>ocamlPackages.ocsigen_server</li>
    <li>ocamlPackages.ocsipersist</li>
    <li>ocamlPackages.ocsipersist-lib</li>
    <li>ocamlPackages.ocsipersist-pgsql</li>
    <li>ocamlPackages.ocsipersist-sqlite</li>
    <li>ocamlPackages.ocurl</li>
    <li>ocamlPackages.odoc</li>
    <li>ocamlPackages.opium</li>
    <li>ocamlPackages.opium_kernel</li>
    <li>ocamlPackages.otr</li>
    <li>ocamlPackages.parsexp</li>
    <li>ocamlPackages.patience_diff</li>
    <li>ocamlPackages.pcap-format</li>
    <li>ocamlPackages.pgocaml</li>
    <li>ocamlPackages.pgocaml_ppx</li>
    <li>ocamlPackages.phylogenetics</li>
    <li>ocamlPackages.piqi</li>
    <li>ocamlPackages.piqi-ocaml</li>
    <li>ocamlPackages.posixat</li>
    <li>ocamlPackages.pp</li>
    <li>ocamlPackages.ppx_accessor</li>
    <li>ocamlPackages.ppx_assert</li>
    <li>ocamlPackages.ppx_bap</li>
    <li>ocamlPackages.ppx_base</li>
    <li>ocamlPackages.ppx_bench</li>
    <li>ocamlPackages.ppx_bin_prot</li>
    <li>ocamlPackages.ppx_bitstring</li>
    <li>ocamlPackages.ppx_blob</li>
    <li>ocamlPackages.ppx_cold</li>
    <li>ocamlPackages.ppx_compare</li>
    <li>ocamlPackages.ppx_css</li>
    <li>ocamlPackages.ppx_cstruct</li>
    <li>ocamlPackages.ppx_cstubs</li>
    <li>ocamlPackages.ppx_custom_printf</li>
    <li>ocamlPackages.ppx_deriving</li>
    <li>ocamlPackages.ppx_deriving_cmdliner</li>
    <li>ocamlPackages.ppx_deriving_protobuf</li>
    <li>ocamlPackages.ppx_deriving_rpc</li>
    <li>ocamlPackages.ppx_deriving_yaml</li>
    <li>ocamlPackages.ppx_deriving_yojson</li>
    <li>ocamlPackages.ppx_disable_unused_warnings</li>
    <li>ocamlPackages.ppx_enumerate</li>
    <li>ocamlPackages.ppx_expect</li>
    <li>ocamlPackages.ppx_fields_conv</li>
    <li>ocamlPackages.ppx_fixed_literal</li>
    <li>ocamlPackages.ppx_gen_rec</li>
    <li>ocamlPackages.ppx_hash</li>
    <li>ocamlPackages.ppx_here</li>
    <li>ocamlPackages.ppx_ignore_instrumentation</li>
    <li>ocamlPackages.ppx_import</li>
    <li>ocamlPackages.ppx_inline_test</li>
    <li>ocamlPackages.ppx_irmin</li>
    <li>ocamlPackages.ppx_jane</li>
    <li>ocamlPackages.ppx_js_style</li>
    <li>ocamlPackages.ppx_let</li>
    <li>ocamlPackages.ppx_log</li>
    <li>ocamlPackages.ppx_module_timer</li>
    <li>ocamlPackages.ppx_optcomp</li>
    <li>ocamlPackages.ppx_optional</li>
    <li>ocamlPackages.ppx_pattern_bind</li>
    <li>ocamlPackages.ppx_pipebang</li>
    <li>ocamlPackages.ppx_python</li>
    <li>ocamlPackages.ppx_repr</li>
    <li>ocamlPackages.ppx_sexp_conv</li>
    <li>ocamlPackages.ppx_sexp_message</li>
    <li>ocamlPackages.ppx_sexp_value</li>
    <li>ocamlPackages.ppx_stable</li>
    <li>ocamlPackages.ppx_string</li>
    <li>ocamlPackages.ppx_typed_fields</li>
    <li>ocamlPackages.ppx_typerep_conv</li>
    <li>ocamlPackages.ppx_variants_conv</li>
    <li>ocamlPackages.ppxlib</li>
    <li>ocamlPackages.prettym</li>
    <li>ocamlPackages.printbox</li>
    <li>ocamlPackages.printbox-text</li>
    <li>ocamlPackages.profunctor</li>
    <li>ocamlPackages.progress</li>
    <li>ocamlPackages.protocol_version_header</li>
    <li>ocamlPackages.ptime</li>
    <li>ocamlPackages.pythonlib</li>
    <li>ocamlPackages.re2</li>
    <li>ocamlPackages.re2_stable</li>
    <li>ocamlPackages.reason</li>
    <li>ocamlPackages.reason-native.console</li>
    <li>ocamlPackages.reason-native.dir</li>
    <li>ocamlPackages.reason-native.file-context-printer</li>
    <li>ocamlPackages.reason-native.fp</li>
    <li>ocamlPackages.reason-native.pastel</li>
    <li>ocamlPackages.reason-native.pastel-console</li>
    <li>ocamlPackages.reason-native.refmterr</li>
    <li>ocamlPackages.reason-native.rely</li>
    <li>ocamlPackages.reason-native.rely-junit-reporter</li>
    <li>ocamlPackages.rebez</li>
    <li>ocamlPackages.record_builder</li>
    <li>ocamlPackages.reperf</li>
    <li>ocamlPackages.repr</li>
    <li>ocamlPackages.resource_cache</li>
    <li>ocamlPackages.resto</li>
    <li>ocamlPackages.resto-acl</li>
    <li>ocamlPackages.resto-cohttp</li>
    <li>ocamlPackages.resto-cohttp-client</li>
    <li>ocamlPackages.resto-cohttp-self-serving-client</li>
    <li>ocamlPackages.resto-cohttp-server</li>
    <li>ocamlPackages.resto-directory</li>
    <li>ocamlPackages.resto-json</li>
    <li>ocamlPackages.rpclib</li>
    <li>ocamlPackages.rpclib-lwt</li>
    <li>ocamlPackages.secp256k1</li>
    <li>ocamlPackages.sedlex</li>
    <li>ocamlPackages.sexp</li>
    <li>ocamlPackages.sexp_diff</li>
    <li>ocamlPackages.sexp_macro</li>
    <li>ocamlPackages.sexp_pretty</li>
    <li>ocamlPackages.sexp_select</li>
    <li>ocamlPackages.sexplib</li>
    <li>ocamlPackages.sexplib0</li>
    <li>ocamlPackages.shared-memory-ring</li>
    <li>ocamlPackages.shared-memory-ring-lwt</li>
    <li>ocamlPackages.shell</li>
    <li>ocamlPackages.shexp</li>
    <li>ocamlPackages.spawn</li>
    <li>ocamlPackages.splay_tree</li>
    <li>ocamlPackages.splittable_random</li>
    <li>ocamlPackages.stdio</li>
    <li>ocamlPackages.syslog-message</li>
    <li>ocamlPackages.tar</li>
    <li>ocamlPackages.tar-unix</li>
    <li>ocamlPackages.telegraml</li>
    <li>ocamlPackages.textutils</li>
    <li>ocamlPackages.textutils_kernel</li>
    <li>ocamlPackages.time_now</li>
    <li>ocamlPackages.timezone</li>
    <li>ocamlPackages.tls</li>
    <li>ocamlPackages.tls-async</li>
    <li>ocamlPackages.tls-mirage</li>
    <li>ocamlPackages.topological_sort</li>
    <li>ocamlPackages.tsort</li>
    <li>ocamlPackages.tuntap</li>
    <li>ocamlPackages.typerep</li>
    <li>ocamlPackages.unstrctrd</li>
    <li>ocamlPackages.uri</li>
    <li>ocamlPackages.uri-sexp</li>
    <li>ocamlPackages.uuuu</li>
    <li>ocamlPackages.variantslib</li>
    <li>ocamlPackages.vcaml</li>
    <li>ocamlPackages.vchan</li>
    <li>ocamlPackages.vg</li>
    <li>ocamlPackages.virtual_dom</li>
    <li>ocamlPackages.visitors</li>
    <li>ocamlPackages.wayland</li>
    <li>ocamlPackages.webbrowser</li>
    <li>ocamlPackages.webmachine</li>
    <li>ocamlPackages.x509</li>
    <li>ocamlPackages.xenstore</li>
    <li>ocamlPackages.xenstore-tool</li>
    <li>ocamlPackages.xenstore_transport</li>
    <li>ocamlPackages.xtmpl</li>
    <li>ocamlPackages.xtmpl_ppx</li>
    <li>ocamlPackages.yaml</li>
    <li>ocamlPackages.yaml-sexp</li>
    <li>ocamlPackages.zarith_stubs_js</li>
    <li>ocamlPackages.zstandard</li>
    <li>ocamlformat (ocamlformat_0_23_0)</li>
    <li>ocamlformat_0_19_0</li>
    <li>ocamlformat_0_20_0</li>
    <li>ocamlformat_0_20_1</li>
    <li>ocamlformat_0_21_0</li>
    <li>ocamlformat_0_22_4</li>
    <li>patdiff</li>
    <li>python310Packages.bap</li>
    <li>python39Packages.bap</li>
    <li>satysfi</li>
    <li>soupault</li>
    <li>stog</li>
    <li>why3</li>
  </ul>
</details>

symphorien commented:
I cannot find a build on hydra of, say, carton-lwt so I cannot know if it's a regression or if it's only broken on darwin. Plus you don't give logs, so it's net very actionable.

siraben commented:
@symphorien I should really use a variant of `nixpkgs-review` that gives logs... Let me re-run.

siraben commented:
Result of `nixpkgs-review pr 166033` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>flitter</li>
    <li>ocamlPackages.ocaml-r</li>
    <li>ocamlPackages.reason-native.qcheck-rely</li>
    <li>ocamlPackages.torch</li>
    <li>ocamlPackages.wodan</li>
    <li>ocamlPackages.wodan-irmin</li>
    <li>ocamlPackages.wodan-unix</li>
  </ul>
</details>
<details>
  <summary>49 packages failed to build:</summary>
  <ul>
    <li>coqPackages.serapi</li>
    <li>haxe</li>
    <li>haxePackages.format</li>
    <li>haxePackages.heaps</li>
    <li>haxePackages.hlopenal</li>
    <li>haxePackages.hlsdl</li>
    <li>haxePackages.hxcpp</li>
    <li>haxePackages.hxcs</li>
    <li>haxePackages.hxjava</li>
    <li>haxePackages.hxnodejs_4</li>
    <li>jackline</li>
    <li>ocamlPackages.arp</li>
    <li>ocamlPackages.async_rpc_websocket</li>
    <li>ocamlPackages.bonsai</li>
    <li>ocamlPackages.caqti-driver-mariadb</li>
    <li>ocamlPackages.carton</li>
    <li>ocamlPackages.carton-git</li>
    <li>ocamlPackages.carton-lwt</li>
    <li>ocamlPackages.cohttp-async</li>
    <li>ocamlPackages.cohttp-mirage</li>
    <li>ocamlPackages.cohttp_async_websocket</li>
    <li>ocamlPackages.conduit-mirage</li>
    <li>ocamlPackages.dns-certify</li>
    <li>ocamlPackages.dns-cli</li>
    <li>ocamlPackages.dns-client</li>
    <li>ocamlPackages.dns-mirage</li>
    <li>ocamlPackages.dns-resolver</li>
    <li>ocamlPackages.dns-server</li>
    <li>ocamlPackages.dns-stub</li>
    <li>ocamlPackages.git</li>
    <li>ocamlPackages.git-cohttp</li>
    <li>ocamlPackages.git-cohttp-unix</li>
    <li>ocamlPackages.git-paf</li>
    <li>ocamlPackages.git-unix</li>
    <li>ocamlPackages.irmin-containers</li>
    <li>ocamlPackages.irmin-git</li>
    <li>ocamlPackages.irmin-http</li>
    <li>ocamlPackages.irmin-mirage-git</li>
    <li>ocamlPackages.irmin-mirage-graphql</li>
    <li>ocamlPackages.irmin-unix</li>
    <li>ocamlPackages.mirage-nat</li>
    <li>ocamlPackages.mrmime</li>
    <li>ocamlPackages.paf</li>
    <li>ocamlPackages.paf-cohttp</li>
    <li>ocamlPackages.paf-le</li>
    <li>ocamlPackages.piaf</li>
    <li>ocamlPackages.rosetta</li>
    <li>ocamlPackages.tcpip</li>
    <li>wayland-proxy-virtwl</li>
  </ul>
</details>
<details>
  <summary>397 packages built:</summary>
  <ul>
    <li>acgtk</li>
    <li>beluga</li>
    <li>comby</li>
    <li>coqPackages.coq-elpi</li>
    <li>coqPackages.graph-theory</li>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.mathcomp-analysis</li>
    <li>coqPackages.trakt</li>
    <li>dune-release</li>
    <li>easycrypt</li>
    <li>eff</li>
    <li>flow</li>
    <li>framac</li>
    <li>fstar</li>
    <li>libbap</li>
    <li>ocaml-crunch</li>
    <li>ocamlPackages.abstract_algebra</li>
    <li>ocamlPackages.accessor</li>
    <li>ocamlPackages.accessor_async</li>
    <li>ocamlPackages.accessor_base</li>
    <li>ocamlPackages.accessor_core</li>
    <li>ocamlPackages.alcotest-lwt</li>
    <li>ocamlPackages.alcotest-mirage</li>
    <li>ocamlPackages.angstrom</li>
    <li>ocamlPackages.angstrom-async</li>
    <li>ocamlPackages.angstrom-lwt-unix</li>
    <li>ocamlPackages.angstrom-unix</li>
    <li>ocamlPackages.asn1-combinators</li>
    <li>ocamlPackages.async</li>
    <li>ocamlPackages.async_extra</li>
    <li>ocamlPackages.async_find</li>
    <li>ocamlPackages.async_interactive</li>
    <li>ocamlPackages.async_js</li>
    <li>ocamlPackages.async_kernel</li>
    <li>ocamlPackages.async_rpc_kernel</li>
    <li>ocamlPackages.async_sendfile</li>
    <li>ocamlPackages.async_shell</li>
    <li>ocamlPackages.async_ssl</li>
    <li>ocamlPackages.async_unix</li>
    <li>ocamlPackages.async_websocket</li>
    <li>ocamlPackages.awa</li>
    <li>ocamlPackages.awa-lwt</li>
    <li>ocamlPackages.awa-mirage</li>
    <li>ocamlPackages.bap</li>
    <li>ocamlPackages.base</li>
    <li>ocamlPackages.base64</li>
    <li>ocamlPackages.base_bigstring</li>
    <li>ocamlPackages.base_quickcheck</li>
    <li>ocamlPackages.bigarray-overlap</li>
    <li>ocamlPackages.bignum</li>
    <li>ocamlPackages.bin_prot</li>
    <li>ocamlPackages.biocaml</li>
    <li>ocamlPackages.bisect_ppx</li>
    <li>ocamlPackages.bistro</li>
    <li>ocamlPackages.bls12-381-unix</li>
    <li>ocamlPackages.bos</li>
    <li>ocamlPackages.brisk-reconciler</li>
    <li>ocamlPackages.ca-certs</li>
    <li>ocamlPackages.ca-certs-nss</li>
    <li>ocamlPackages.camlimages</li>
    <li>ocamlPackages.caqti</li>
    <li>ocamlPackages.caqti-async</li>
    <li>ocamlPackages.caqti-driver-postgresql</li>
    <li>ocamlPackages.caqti-driver-sqlite3</li>
    <li>ocamlPackages.caqti-dynload</li>
    <li>ocamlPackages.caqti-lwt</li>
    <li>ocamlPackages.caqti-type-calendar</li>
    <li>ocamlPackages.cfstream</li>
    <li>ocamlPackages.checkseum</li>
    <li>ocamlPackages.cinaps</li>
    <li>ocamlPackages.cohttp</li>
    <li>ocamlPackages.cohttp-lwt</li>
    <li>ocamlPackages.cohttp-lwt-unix</li>
    <li>ocamlPackages.conduit</li>
    <li>ocamlPackages.conduit-async</li>
    <li>ocamlPackages.conduit-lwt</li>
    <li>ocamlPackages.conduit-lwt-unix</li>
    <li>ocamlPackages.containers</li>
    <li>ocamlPackages.containers-data</li>
    <li>ocamlPackages.cooltt</li>
    <li>ocamlPackages.core</li>
    <li>ocamlPackages.core_bench</li>
    <li>ocamlPackages.core_extended</li>
    <li>ocamlPackages.core_kernel</li>
    <li>ocamlPackages.core_unix</li>
    <li>ocamlPackages.cow</li>
    <li>ocamlPackages.crunch</li>
    <li>ocamlPackages.cstruct-async</li>
    <li>ocamlPackages.cstruct-sexp</li>
    <li>ocamlPackages.csvfields</li>
    <li>ocamlPackages.dap</li>
    <li>ocamlPackages.data-encoding</li>
    <li>ocamlPackages.dbf</li>
    <li>ocamlPackages.decompress</li>
    <li>ocamlPackages.delimited_parsing</li>
    <li>ocamlPackages.digestif</li>
    <li>ocamlPackages.dns</li>
    <li>ocamlPackages.dns-tsig</li>
    <li>ocamlPackages.dose3</li>
    <li>ocamlPackages.dune-rpc</li>
    <li>ocamlPackages.ecaml</li>
    <li>ocamlPackages.eliom</li>
    <li>ocamlPackages.elpi</li>
    <li>ocamlPackages.emile</li>
    <li>ocamlPackages.encore</li>
    <li>ocamlPackages.erm_xmpp</li>
    <li>ocamlPackages.ethernet</li>
    <li>ocamlPackages.expect_test_helpers_async</li>
    <li>ocamlPackages.expect_test_helpers_core</li>
    <li>ocamlPackages.ezjsonm</li>
    <li>ocamlPackages.faraday-async</li>
    <li>ocamlPackages.fieldslib</li>
    <li>ocamlPackages.flex</li>
    <li>ocamlPackages.functoria</li>
    <li>ocamlPackages.functoria-runtime</li>
    <li>ocamlPackages.fuzzy_match</li>
    <li>ocamlPackages.gapi_ocaml</li>
    <li>ocamlPackages.gen_js_api</li>
    <li>ocamlPackages.genspio</li>
    <li>ocamlPackages.graphql-cohttp</li>
    <li>ocamlPackages.graphql_ppx</li>
    <li>ocamlPackages.h2</li>
    <li>ocamlPackages.hack_parallel</li>
    <li>ocamlPackages.happy-eyeballs</li>
    <li>ocamlPackages.higher_kinded</li>
    <li>ocamlPackages.higlo</li>
    <li>ocamlPackages.hpack</li>
    <li>ocamlPackages.httpaf</li>
    <li>ocamlPackages.imagelib</li>
    <li>ocamlPackages.incr_dom</li>
    <li>ocamlPackages.incr_map</li>
    <li>ocamlPackages.incr_select</li>
    <li>ocamlPackages.incremental</li>
    <li>ocamlPackages.index</li>
    <li>ocamlPackages.int_repr</li>
    <li>ocamlPackages.ipaddr</li>
    <li>ocamlPackages.ipaddr-cstruct</li>
    <li>ocamlPackages.ipaddr-sexp</li>
    <li>ocamlPackages.iri</li>
    <li>ocamlPackages.irmin</li>
    <li>ocamlPackages.irmin-chunk</li>
    <li>ocamlPackages.irmin-fs</li>
    <li>ocamlPackages.irmin-graphql</li>
    <li>ocamlPackages.irmin-layers</li>
    <li>ocamlPackages.irmin-mirage</li>
    <li>ocamlPackages.irmin-pack</li>
    <li>ocamlPackages.irmin-test</li>
    <li>ocamlPackages.irmin-watcher</li>
    <li>ocamlPackages.iter</li>
    <li>ocamlPackages.jane-street-headers</li>
    <li>ocamlPackages.jingoo</li>
    <li>ocamlPackages.js_of_ocaml</li>
    <li>ocamlPackages.js_of_ocaml-compiler</li>
    <li>ocamlPackages.js_of_ocaml-lwt</li>
    <li>ocamlPackages.js_of_ocaml-ppx</li>
    <li>ocamlPackages.js_of_ocaml-ppx_deriving_json</li>
    <li>ocamlPackages.js_of_ocaml-tyxml</li>
    <li>ocamlPackages.json-data-encoding</li>
    <li>ocamlPackages.json-data-encoding-bson</li>
    <li>ocamlPackages.jsonaf</li>
    <li>ocamlPackages.jst-config</li>
    <li>ocamlPackages.junit</li>
    <li>ocamlPackages.junit_alcotest</li>
    <li>ocamlPackages.junit_ounit</li>
    <li>ocamlPackages.jwto</li>
    <li>ocamlPackages.lens</li>
    <li>ocamlPackages.letsencrypt</li>
    <li>ocamlPackages.letsencrypt-app</li>
    <li>ocamlPackages.letsencrypt-dns</li>
    <li>ocamlPackages.logs</li>
    <li>ocamlPackages.lsp</li>
    <li>ocamlPackages.lwt-exit</li>
    <li>ocamlPackages.lwt_ppx</li>
    <li>ocamlPackages.macaddr</li>
    <li>ocamlPackages.macaddr-cstruct</li>
    <li>ocamlPackages.macaddr-sexp</li>
    <li>ocamlPackages.macaque</li>
    <li>ocamlPackages.mdx</li>
    <li>ocamlPackages.metrics-lwt</li>
    <li>ocamlPackages.metrics-mirage</li>
    <li>ocamlPackages.metrics-unix</li>
    <li>ocamlPackages.mimic</li>
    <li>ocamlPackages.mirage</li>
    <li>ocamlPackages.mirage-block-combinators</li>
    <li>ocamlPackages.mirage-block-unix</li>
    <li>ocamlPackages.mirage-bootvar-xen</li>
    <li>ocamlPackages.mirage-channel</li>
    <li>ocamlPackages.mirage-crypto-ec</li>
    <li>ocamlPackages.mirage-crypto-pk</li>
    <li>ocamlPackages.mirage-crypto-rng</li>
    <li>ocamlPackages.mirage-crypto-rng-async</li>
    <li>ocamlPackages.mirage-crypto-rng-mirage</li>
    <li>ocamlPackages.mirage-flow-combinators</li>
    <li>ocamlPackages.mirage-flow-unix</li>
    <li>ocamlPackages.mirage-logs</li>
    <li>ocamlPackages.mirage-net</li>
    <li>ocamlPackages.mirage-net-xen</li>
    <li>ocamlPackages.mirage-profile</li>
    <li>ocamlPackages.mirage-protocols</li>
    <li>ocamlPackages.mirage-runtime</li>
    <li>ocamlPackages.mirage-stack</li>
    <li>ocamlPackages.mirage-types</li>
    <li>ocamlPackages.mirage-types-lwt</li>
    <li>ocamlPackages.mirage-unix</li>
    <li>ocamlPackages.mirage-vnetif</li>
    <li>ocamlPackages.mirage-xen</li>
    <li>ocamlPackages.mldoc</li>
    <li>ocamlPackages.mtime</li>
    <li>ocamlPackages.mustache</li>
    <li>ocamlPackages.netchannel</li>
    <li>ocamlPackages.ocaml-compiler-libs</li>
    <li>ocamlPackages.ocaml-embed-file</li>
    <li>ocamlPackages.ocaml-lsp</li>
    <li>ocamlPackages.ocaml-monadic</li>
    <li>ocamlPackages.ocaml-print-intf</li>
    <li>ocamlPackages.ocaml_intrinsics</li>
    <li>ocamlPackages.ocamlformat-rpc-lib</li>
    <li>ocamlPackages.ocf_ppx</li>
    <li>ocamlPackages.ocplib-json-typed</li>
    <li>ocamlPackages.ocplib-json-typed-browser</li>
    <li>ocamlPackages.ocplib-json-typed-bson</li>
    <li>ocamlPackages.ocsigen-ppx-rpc</li>
    <li>ocamlPackages.ocsigen-start</li>
    <li>ocamlPackages.ocsigen-toolkit</li>
    <li>ocamlPackages.ocsigen_server</li>
    <li>ocamlPackages.ocsipersist</li>
    <li>ocamlPackages.ocsipersist-lib</li>
    <li>ocamlPackages.ocsipersist-pgsql</li>
    <li>ocamlPackages.ocsipersist-sqlite</li>
    <li>ocamlPackages.ocurl</li>
    <li>ocamlPackages.odoc</li>
    <li>ocamlPackages.opium</li>
    <li>ocamlPackages.opium_kernel</li>
    <li>ocamlPackages.otr</li>
    <li>ocamlPackages.parsexp</li>
    <li>ocamlPackages.patience_diff</li>
    <li>ocamlPackages.pcap-format</li>
    <li>ocamlPackages.pgocaml</li>
    <li>ocamlPackages.pgocaml_ppx</li>
    <li>ocamlPackages.phylogenetics</li>
    <li>ocamlPackages.piqi</li>
    <li>ocamlPackages.piqi-ocaml</li>
    <li>ocamlPackages.posixat</li>
    <li>ocamlPackages.pp</li>
    <li>ocamlPackages.ppx_accessor</li>
    <li>ocamlPackages.ppx_assert</li>
    <li>ocamlPackages.ppx_bap</li>
    <li>ocamlPackages.ppx_base</li>
    <li>ocamlPackages.ppx_bench</li>
    <li>ocamlPackages.ppx_bin_prot</li>
    <li>ocamlPackages.ppx_bitstring</li>
    <li>ocamlPackages.ppx_blob</li>
    <li>ocamlPackages.ppx_cold</li>
    <li>ocamlPackages.ppx_compare</li>
    <li>ocamlPackages.ppx_css</li>
    <li>ocamlPackages.ppx_cstruct</li>
    <li>ocamlPackages.ppx_cstubs</li>
    <li>ocamlPackages.ppx_custom_printf</li>
    <li>ocamlPackages.ppx_deriving</li>
    <li>ocamlPackages.ppx_deriving_cmdliner</li>
    <li>ocamlPackages.ppx_deriving_protobuf</li>
    <li>ocamlPackages.ppx_deriving_rpc</li>
    <li>ocamlPackages.ppx_deriving_yaml</li>
    <li>ocamlPackages.ppx_deriving_yojson</li>
    <li>ocamlPackages.ppx_disable_unused_warnings</li>
    <li>ocamlPackages.ppx_enumerate</li>
    <li>ocamlPackages.ppx_expect</li>
    <li>ocamlPackages.ppx_fields_conv</li>
    <li>ocamlPackages.ppx_fixed_literal</li>
    <li>ocamlPackages.ppx_gen_rec</li>
    <li>ocamlPackages.ppx_hash</li>
    <li>ocamlPackages.ppx_here</li>
    <li>ocamlPackages.ppx_ignore_instrumentation</li>
    <li>ocamlPackages.ppx_import</li>
    <li>ocamlPackages.ppx_inline_test</li>
    <li>ocamlPackages.ppx_irmin</li>
    <li>ocamlPackages.ppx_jane</li>
    <li>ocamlPackages.ppx_js_style</li>
    <li>ocamlPackages.ppx_let</li>
    <li>ocamlPackages.ppx_log</li>
    <li>ocamlPackages.ppx_module_timer</li>
    <li>ocamlPackages.ppx_optcomp</li>
    <li>ocamlPackages.ppx_optional</li>
    <li>ocamlPackages.ppx_pattern_bind</li>
    <li>ocamlPackages.ppx_pipebang</li>
    <li>ocamlPackages.ppx_python</li>
    <li>ocamlPackages.ppx_repr</li>
    <li>ocamlPackages.ppx_sexp_conv</li>
    <li>ocamlPackages.ppx_sexp_value</li>
    <li>ocamlPackages.ppx_stable</li>
    <li>ocamlPackages.ppx_string</li>
    <li>ocamlPackages.ppx_typed_fields</li>
    <li>ocamlPackages.ppx_typerep_conv</li>
    <li>ocamlPackages.ppx_variants_conv</li>
    <li>ocamlPackages.ppxlib</li>
    <li>ocamlPackages.prettym</li>
    <li>ocamlPackages.printbox</li>
    <li>ocamlPackages.printbox-text</li>
    <li>ocamlPackages.profunctor</li>
    <li>ocamlPackages.progress</li>
    <li>ocamlPackages.protocol_version_header</li>
    <li>ocamlPackages.ptime</li>
    <li>ocamlPackages.pythonlib</li>
    <li>ocamlPackages.re2</li>
    <li>ocamlPackages.re2_stable</li>
    <li>ocamlPackages.reason</li>
    <li>ocamlPackages.reason-native.console</li>
    <li>ocamlPackages.reason-native.dir</li>
    <li>ocamlPackages.reason-native.file-context-printer</li>
    <li>ocamlPackages.reason-native.fp</li>
    <li>ocamlPackages.reason-native.pastel</li>
    <li>ocamlPackages.reason-native.pastel-console</li>
    <li>ocamlPackages.reason-native.refmterr</li>
    <li>ocamlPackages.reason-native.rely</li>
    <li>ocamlPackages.reason-native.rely-junit-reporter</li>
    <li>ocamlPackages.rebez</li>
    <li>ocamlPackages.record_builder</li>
    <li>ocamlPackages.reperf</li>
    <li>ocamlPackages.repr</li>
    <li>ocamlPackages.resource_cache</li>
    <li>ocamlPackages.resto</li>
    <li>ocamlPackages.resto-acl</li>
    <li>ocamlPackages.resto-cohttp</li>
    <li>ocamlPackages.resto-cohttp-client</li>
    <li>ocamlPackages.resto-cohttp-self-serving-client</li>
    <li>ocamlPackages.resto-cohttp-server</li>
    <li>ocamlPackages.resto-directory</li>
    <li>ocamlPackages.resto-json</li>
    <li>ocamlPackages.rpclib</li>
    <li>ocamlPackages.rpclib-lwt</li>
    <li>ocamlPackages.secp256k1</li>
    <li>ocamlPackages.sedlex</li>
    <li>ocamlPackages.sexp</li>
    <li>ocamlPackages.sexp_diff</li>
    <li>ocamlPackages.sexp_macro</li>
    <li>ocamlPackages.sexp_pretty</li>
    <li>ocamlPackages.sexp_select</li>
    <li>ocamlPackages.sexplib</li>
    <li>ocamlPackages.sexplib0</li>
    <li>ocamlPackages.shared-memory-ring</li>
    <li>ocamlPackages.shared-memory-ring-lwt</li>
    <li>ocamlPackages.shell</li>
    <li>ocamlPackages.shexp</li>
    <li>ocamlPackages.spawn</li>
    <li>ocamlPackages.splay_tree</li>
    <li>ocamlPackages.splittable_random</li>
    <li>ocamlPackages.stdio</li>
    <li>ocamlPackages.tar</li>
    <li>ocamlPackages.tar-unix</li>
    <li>ocamlPackages.telegraml</li>
    <li>ocamlPackages.textutils</li>
    <li>ocamlPackages.textutils_kernel</li>
    <li>ocamlPackages.time_now</li>
    <li>ocamlPackages.timezone</li>
    <li>ocamlPackages.tls</li>
    <li>ocamlPackages.tls-async</li>
    <li>ocamlPackages.tls-mirage</li>
    <li>ocamlPackages.topological_sort</li>
    <li>ocamlPackages.tsort</li>
    <li>ocamlPackages.tuntap</li>
    <li>ocamlPackages.typerep</li>
    <li>ocamlPackages.unstrctrd</li>
    <li>ocamlPackages.uri</li>
    <li>ocamlPackages.uri-sexp</li>
    <li>ocamlPackages.uuuu</li>
    <li>ocamlPackages.variantslib</li>
    <li>ocamlPackages.vcaml</li>
    <li>ocamlPackages.vchan</li>
    <li>ocamlPackages.vg</li>
    <li>ocamlPackages.virtual_dom</li>
    <li>ocamlPackages.visitors</li>
    <li>ocamlPackages.wayland</li>
    <li>ocamlPackages.webbrowser</li>
    <li>ocamlPackages.webmachine</li>
    <li>ocamlPackages.x509</li>
    <li>ocamlPackages.xenstore</li>
    <li>ocamlPackages.xenstore-tool</li>
    <li>ocamlPackages.xenstore_transport</li>
    <li>ocamlPackages.xtmpl</li>
    <li>ocamlPackages.xtmpl_ppx</li>
    <li>ocamlPackages.yaml</li>
    <li>ocamlPackages.yaml-sexp</li>
    <li>ocamlPackages.zarith_stubs_js</li>
    <li>ocamlPackages.zstandard</li>
    <li>ocamlformat (ocamlformat_0_23_0)</li>
    <li>ocamlformat_0_19_0</li>
    <li>ocamlformat_0_20_0</li>
    <li>ocamlformat_0_20_1</li>
    <li>ocamlformat_0_21_0</li>
    <li>ocamlformat_0_22_4</li>
    <li>patdiff</li>
    <li>python310Packages.bap</li>
    <li>python39Packages.bap</li>
    <li>satysfi</li>
    <li>soupault</li>
    <li>stog</li>
    <li>why3</li>
  </ul>
</details>

symphorien commented:
Compared to x86_64 and to hydra's aarch64-darwin job (which does not contain all of ocamlPackage), the only regression is jackline https://hydra.nixos.org/eval/1771667?filter=jackline&compare=1771622&full=

bcc32 commented:
Thanks for the help, everybody!

bcc32 commented:
Thanks @vbgl.  I patched serapi in #182170 just now.
